tokio-util/src/task/spawn_aborting.rs

```diff
@@ -0,0 +1,38 @@
+use std::ops::{Deref, DerefMut};
+
+use tokio::task::JoinHandle;
+
+use futures_core::Future;
+
+/// This is a wrapper type around JoinHandle that allows it to be dropped.
+#[derive(Debug)]
+pub struct DropHandle<T>(JoinHandle<T>);
```

Comment:
This name is okay, but I think there is precedent for the name `AbortHandle` elsewhere, so we might want to prefer that name.

Reply:
Actually, we have a type called `AbortHandle` in Tokio, and that does something else ... so that name doesn't work.

But another option is `AbortOnDropHandle`. I'll let you pick.